### ISSUE TITLE
Lerna is dead

### ISSUE THREAD

**Issue by rhysd** [neovim/node-client#196]
[Lerna is no longer maintained](https://github.com/lerna/lerna/issues/2703). We would need to migrate to some other tools (external tool like [pnpm](https://github.com/pnpm/pnpm) or yarn/npm workspace). I don't have much experience to manage monorepo so I don't have strong opinion.

**Comment by justinmk** [comment]
Regular NPM sounds great to me. If it can't handle "monorepo" for some reason then I guess yarn is fine, unless pnpm is more popular. Whatever is the most boring (popular) will hopefully give us the least trouble.

**Comment by rhysd** [comment]
I agree that less dependencies is better. I will try both npm workspaces and yarn workspaces.

**Comment by rhysd** [comment]
I tried both [yarn](https://github.com/neovim/node-client/tree/issue-196-yarn) and [npm](https://github.com/neovim/node-client/tree/issue-196-npm), and chose npm at #197.

This is because

- Yarn does not support workspaces out-of-the-box. It requires a Yarn plugin and newer version of Yarn.
- Yarn enables PnP by default, but it works very differently from npm. Maintaining this repository to work with both Yarn and npm will be harder.
- While trying migration to Yarn, I faced [some lock file issue](https://github.com/yarnpkg/yarn/issues/6739). I could avoid it by deleting the local repository and cloning it again, but I don't know why it was fixed. And Yarn's lock file was slightly modified after editing `scripts` section of `package.json`.

I concerned that npm is slower than Yarn. Though Yarn is faster, I think npm is still acceptable.

- Yarn: 2m27s https://github.com/neovim/node-client/actions/runs/2264242857
- npm: 3m50s https://github.com/neovim/node-client/actions/runs/2273441388

Remaining concern is releasing. Since @billyvg is responsible for releases, I don't know how this change affects work for release. I think lerna provides some useful feature for releasing new versions. Now they would not be available.

Steps to make new releases are now:

1. Run `npm version` with [`--workspace` (or `-w`) option](https://docs.npmjs.com/cli/v8/commands/npm-version#workspace)
2. Go to the package directory to release
3. Run `npm publish`

I believe this flow is acceptable since we release packages not so frequently. But I'd like to confirm it with @billyvg.

**Comment by justinmk** [comment]
Why do we need workspaces/ monorepo ?

**Comment by justinmk** [comment]
> This repository needs to be monorepo because we maintain separate two packages.
> 
> `@neovim/decorators` is a package to make a plugin host.

Maybe we should get rid of that. Doesn't seem worth the complexity.
